jobs/resource_job.go

```diff
@@ -200,6 +200,11 @@ type ContinuousConf struct {
 type Queue struct {
 }
 
+type JobRunAs struct {
+	UserName             string `json:"user_name,omitempty"`
+	ServicePrincipalName string `json:"service_principal_name,omitempty"`
```

Comment:
we may need to add a customization to set `ConflictsWith` between these two fields. Also, we may need to add checks that both strings aren't empty...